front/src/common/api/osrdEditoastApi.ts

```diff
@@ -70,6 +80,97 @@ const osrdEditoastApi = generatedEditoastApi
         },
         providesTags: ['timetable'],
       }),
+      getTimetableItemById: builder.query<TimetableItem, { id: TimetableItemId }>({
+        queryFn: async ({ id: timetableItemId }, { dispatch }) => {
+          let data: TimetableItem;
+          if (isTrainScheduleId(timetableItemId)) {
+            const trainSchedule = await dispatch(
+              generatedEditoastApi.endpoints.getTrainScheduleById.initiate(
+                {
+                  id: extractEditoastIdFromTrainScheduleId(timetableItemId),
+                },
+                { forceRefetch: true, subscribe: false }
+              )
+            ).unwrap();
+            data = { ...trainSchedule, id: timetableItemId };
+          } else {
+            const pacedTrain = await dispatch(
+              generatedEditoastApi.endpoints.getPacedTrainById.initiate(
+                {
+                  id: extractEditoastIdFromPacedTrainId(timetableItemId),
+                },
+                { forceRefetch: true, subscribe: false }
+              )
+            ).unwrap();
+            data = { ...pacedTrain, id: timetableItemId };
+          }
+          return { data };
+        },
+        providesTags: ['timetable', 'train_schedule', 'paced_train'],
+      }),
+      getTrainPath: builder.query<PathfindingResult, { id: TrainId; infraId: number }>({
+        queryFn: async ({ id: trainId, infraId }, { dispatch }) => {
+          let path: PathfindingResult;
+          if (isTrainScheduleId(trainId)) {
+            path = await dispatch(
+              generatedEditoastApi.endpoints.getTrainScheduleByIdPath.initiate(
+                {
+                  id: extractEditoastIdFromTrainScheduleId(trainId),
+                  infraId,
+                },
+                { forceRefetch: true, subscribe: false }
+              )
+            ).unwrap();
+          } else {
+            const pacedTrainId = extractPacedTrainIdFromOccurrenceId(trainId);
+            path = await dispatch(
+              generatedEditoastApi.endpoints.getPacedTrainByIdPath.initiate(
+                {
+                  id: extractEditoastIdFromPacedTrainId(pacedTrainId),
+                  infraId,
+                },
+                { forceRefetch: true, subscribe: false }
+              )
+            ).unwrap();
+          }
+          return { data: path };
+        },
+        providesTags: ['pathfinding', 'train_schedule', 'paced_train'],
```

Comment:
I think we can return the right kind of tag depending on the args here:

```ts
providesTags: (result, error, arg, meta) => isTrainScheduleId(arg.id) ? ['pathfinding', 'train_schedule'] : ['pathfinding', 'paced_train'],
```

Reply:
Done :)